tools/ci-skip-for-maintainers.js

```diff
@@ -33,10 +33,16 @@ const main = async () => {
     pullReqUrl + "/commits",
   );
 
-  commitMessagesMetadata.forEach((commitMessageMetadata) => {
-    // get commit message body
-    commitMessageList.push(commitMessageMetadata["commit"]["message"]);
-  });
+  if (Array.isArray(commitMessagesMetadata)) {
+    commitMessagesMetadata.forEach((commitMessageMetadata) => {
+      // get commit message body
+      commitMessageList.push(commitMessageMetadata["commit"]["message"]);
+    });
+  } else {
+    console.error(
```

Comment:
@zondervancalvez Instead of just logging on stderr we should crash. Otherwise there will be more unintended side-effects to the error.
The bottom line is that if we canot fetch the commit message metadata, then we cannot complete the check at all.

@zondervancalvez Is it possible to refactor this code to use the local git installation and list the commits that way so that we are not hitting the GitHub API at all for this?